fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -132,11 +133,29 @@ public LogicalMetric make(String metricName, String dependentMetric) {
      * @param metricName  Name for the metric we're making
      * @param dependentMetrics  Metrics this metric depends on
      *
-     * @return The new logicalMetric
+     * @return the new logicalMetric
+     *
+     * @deprecated logical metric needs more config-richness to not just configure metric name, but also metric long
+     * name, description, etc. Use {@link #makeInner(LogicalMetricInfo, List)} instead.
      */
+    @Deprecated
     protected abstract LogicalMetric makeInner(String metricName, List<String> dependentMetrics);
 
     /**
+     * Delegated to for actually making the metric.
+     *
+     * @param logicalMetricInfo  Logical metric info provider
+     * @param dependentMetrics  Metrics this metric depends on
+     *
+     * @return the new logicalMetric
+     */
+    protected LogicalMetric makeInner(LogicalMetricInfo logicalMetricInfo, List<String> dependentMetrics) {
```

Comment:
It looks like this was only implemented for `AggregationAverageMaker`, shouldn't it be done for all subclasses of `MetricMaker`?

Reply:
@kevinhinterlong Yes. `AggregationAverageMaker` is for experimenting. All subclasses will be modified